File: client-react/src/constants/user_status.ts
export const USER_STATUS = {
  ACTIVE: 1,
  AWAY: 2,
  DO_NOT_DISTURB: 3,
  INVISIBLE: 4,
} as const;

export const USER_STATUS_2_TEXT = {
  [USER_STATUS.ACTIVE]: 'Active',
  [USER_STATUS.AWAY]: 'Away',
  [USER_STATUS.DO_NOT_DISTURB]: 'Do not disturb',
  [USER_STATUS.INVISIBLE]: 'Invisible',
} as const;

type TUserStatus = typeof USER_STATUS[keyof typeof USER_STATUS];

export type { TUserStatus };

File: server-spring/src/main/java/com/nogenem/skyapp/enums/UserStatus.java
package com.nogenem.skyapp.enums;

import java.util.Arrays;
import java.util.Map;
import java.util.function.Function;
import java.util.stream.Collectors;

import com.fasterxml.jackson.annotation.JsonValue;

import lombok.AllArgsConstructor;
import lombok.Getter;

@AllArgsConstructor
@Getter(onMethod = @__({ @JsonValue }))
public enum UserStatus {
  ACTIVE(1),
  AWAY(2),
  DO_NOT_DISTURB(3),
  INVISIBLE(4);

  private int value;

  private static Map<Integer, UserStatus> reverseLookup = Arrays.stream(UserStatus.values())
      .collect(Collectors.toMap(UserStatus::getValue, Function.identity()));

  public static UserStatus fromInt(final int id) {
    return reverseLookup.getOrDefault(id, ACTIVE);
  }
}
